### PR TITLE
PE-2365: App-Version, App-Platform and App-Platform-Version

### PR DESCRIPTION
**Pull request by matibat**
Adds extra tagging for the Flutter App to distinguish the platform from where it was uploaded.

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/5tufi7h8vct90
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/44njrtdqphdf8

### CODE REVIEW

File: lib/entities/entity.dart [hunk 1]
```diff
@@ -92,4 +99,58 @@ extension TransactionUtils on TransactionBase {
...
     addTag(EntityTag.arFs, '0.11');
   }
+
+  get _platformVersion async {
+    final platform = _platform;
+    final deviceInfoPlugin = DeviceInfoPlugin();
+
+    switch (platform) {
+      case 'Android':
+        final androidDeviceInfo = await deviceInfoPlugin.androidInfo;
```
Comment (karlprieb) [inline lib/entities/entity.dart:109]: Use types here. We know Dart can infer these types, but it's easier to read when we use types.

File: lib/entities/entity.dart [hunk 2]
```diff
@@ -92,4 +99,58 @@ extension TransactionUtils on TransactionBase {
...
+        final systemName = iosDeviceInfo.systemName;
+        final iosVersion = iosDeviceInfo.systemVersion;
+
+        // iOS 13
+        final versionString = '$systemName $iosVersion';
+        return versionString;
+
+      default:
+        // web platform
+
```
Comment (karlprieb) [inline lib/entities/entity.dart:128]: remove empty line

File: lib/entities/entity.dart [hunk 3]
```diff
@@ -92,4 +99,58 @@ extension TransactionUtils on TransactionBase {
...
+        final webDeviceInfo = await deviceInfoPlugin.webBrowserInfo;
+        final browserName = describeEnum(webDeviceInfo.browserName);
+        final browserVersion = webDeviceInfo.appVersion;
+        final versionString = '$browserName $browserVersion';
+        return versionString;
```
Comment (karlprieb) [inline lib/entities/entity.dart:133]: add empty line before to differentiate blocks

File: lib/entities/entity.dart [hunk 4]
```diff
@@ -92,4 +99,58 @@ extension TransactionUtils on TransactionBase {
...
+    final platform = _platform;
+    final deviceInfoPlugin = DeviceInfoPlugin();
+
+    switch (platform) {
+      case 'Android':
+        final androidDeviceInfo = await deviceInfoPlugin.androidInfo;
+        final androidVersion = androidDeviceInfo.version.release;
+        final sdkVersion = androidDeviceInfo.version.sdkInt;
+
+        // 9 (SDK 28)
```
Comment (karlprieb) [inline lib/entities/entity.dart:113]: why did you add this comment?

File: lib/entities/entity.dart [hunk 5]
```diff
@@ -92,4 +99,58 @@ extension TransactionUtils on TransactionBase {
...
+
+    const platform = LocalPlatform();
+
+    /// A string (linux, macos, windows, android, ios, or fuchsia) representing the operating system.
+    final operatingSystem = platform.operatingSystem;
+
+    switch (operatingSystem) {
+      case 'android':
+        return 'Android';
+      case 'ios':
```
Comment (karlprieb) [inline lib/entities/entity.dart:150]: when iOS it can be both iOS or iPadOS, right? Can we use `deviceInfoPlugin.iosInfo.systemName` to differentiate?

Comment (karlprieb) [reply lib/entities/entity.dart:150]: We can start with just iOS for now

File: lib/entities/entity.dart (hunk 2 above)
Comment (matibat) [reply lib/entities/entity.dart:128]: The auto-format did that 😅 I might use `vi` to edit it, but if another dev edits this file and saves with auto formatting, then they would have the same behavior x.x

File: lib/entities/entity.dart (hunk 4 above)
Comment (matibat) [reply lib/entities/entity.dart:113]: I'll update these like this
```suggestion
        // E.g. 9 (SDK 28)
```

Comment (matibat) [reply lib/entities/entity.dart:113]: That's an example only

File: lib/entities/entity.dart (hunk 2 above)
Comment (karlprieb) [reply lib/entities/entity.dart:128]: Ok :D haha

File: lib/entities/entity.dart (hunk 5 above)
Comment (matibat) [reply lib/entities/entity.dart:150]: That's what I'm putting in the App-Platform-Version.
E.g.:
- App-Platform: `iOS`
- App-Platfrom-Version: `iPadOS 15.7`

``` dart
case 'iOS':
        final iosDeviceInfo = await deviceInfoPlugin.iosInfo;
        final systemName = iosDeviceInfo.systemName;
        final iosVersion = iosDeviceInfo.systemVersion;

        // iOS 13
        final versionString = '$systemName $iosVersion';
        return versionString;
```

File: lib/entities/entity.dart (hunk 1 above)
Comment (matibat) [reply lib/entities/entity.dart:109]: Uses explicit types: d30b549e7c791dc601956e8bb377b47a4456d58e

File: lib/entities/entity.dart (hunk 2 above)
Comment (matibat) [reply lib/entities/entity.dart:123]: Replaces by an in-line comment: d30b549e7c791dc601956e8bb377b47a4456d58e

File: lib/entities/entity.dart (hunk 3 above)
Comment (matibat) [reply lib/entities/entity.dart:130]: Places a newline before the return line on each sub-block of the switch: d30b549e7c791dc601956e8bb377b47a4456d58e

File: lib/entities/entity.dart (hunk 4 above)
Comment (matibat) [reply lib/entities/entity.dart]: Removes the example comments: d30b549e7c791dc601956e8bb377b47a4456d58e

File: lib/entities/entity.dart [hunk 6]
```diff
@@ -78,9 +81,13 @@ class EntityTransactionDataNetworkException implements Exception {
...
-  /// TODO: Split App-Name into App-Name and Client
-  void addApplicationTags({required String version, DateTime? unixTime}) {
-    addTag(EntityTag.appName, 'ArDrive-Web');
+  Future<void> addApplicationTags({
+    required String version,
+    DateTime? unixTime,
+  }) async {
+    addTag(EntityTag.appName, 'ArDrive-App');
+    addTag(EntityTag.appPlatform, _platform);
+    addTag(EntityTag.appPlatformVersion, await _platformVersion);
```
[the remark this answers is not in the document]
Comment (matibat) [reply lib/entities/entity.dart:90]: If we do that, we would be still un-blocking PE-2028 and we can un-comment it after we made the decision in a following up tech_ticket

[the remark this answers is not in the document]
Comment (karlprieb) [reply lib/entities/entity.dart:90]: Make sense

[the remark this answers is not in the document]
Comment (matibat) [reply lib/entities/entity.dart]: [895a09f](https://github.com/ardriveapp/ardrive-web/pull/707/commits/895a09ffd98ff48b4db1e64d2bb00eaa9edb9ce1)

**Comment by matibat** [comment]
This PR has created some corrupted bundles when uploading files. Tested on Web, iOS and Android.

From Chrome - 634.79 KB: https://viewblock.io/arweave/tx/KTx4dnUyMi9KPwOWagYkvDti45p-UU6_gICs5Xh6F7g
From Android 11 - 2.03 MB: https://viewblock.io/arweave/tx/GckFW-7rmNDRDZUQbsC7s4Eiu06QtaSvk3phB7dUUtQ
From iPad - 251.11 KB: https://viewblock.io/arweave/tx/H_3Gb0C9F91i0YWIpTG6iwpJLR28mThyrI2Tcuql2Yo

----------

Also, there are some failing GH Actions: a reminder to take a look

**Comment by matibat** [comment]
The corrupted bundles issue is now fixed.

The broken tests are regarding the unsupported platforms; we need to stub those values.

File: lib/blocs/fs_entry_move/fs_entry_move_bloc.dart [hunk 7]
```diff
@@ -40,6 +44,8 @@ class FsEntryMoveBloc extends Bloc<FsEntryMoveEvent, FsEntryMoveState> {
 
     on<FsEntryMoveEvent>(
       (event, emit) async {
+        print('The move event: $event');
```
Comment (matibat) [inline lib/blocs/fs_entry_move/fs_entry_move_bloc.dart:47]: Remove debugging print

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM

File: lib/blocs/fs_entry_move/fs_entry_move_bloc.dart (hunk 7 above)
Comment (matibat) [reply lib/blocs/fs_entry_move/fs_entry_move_bloc.dart]: f2d89da44d0e39de2fbe9efb2691df079b050cdb

**Review by arielmelendez — DISMISSED** [review]

File: lib/utils/app_platform.dart [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  /// A string (linux, macos, windows, android, ios, or fuchsia) representing the operating system.
+  final String operatingSystem = platform.operatingSystem;
+
+  switch (operatingSystem) {
+    case 'android':
+      return 'Android';
+    case 'ios':
+      return 'iOS';
+    default:
+      throw Exception('Unsupported platform $operatingSystem!');
```
Comment (matibat) [inline lib/utils/app_platform.dart:18]: Do we want to throw or to return "unknown"?

**Comment by matibat** [comment]
Tech debt: https://ardrive.atlassian.net/browse/PE-2388

File: lib/utils/app_platform.dart (hunk 8 above)
Comment (matibat) [reply lib/utils/app_platform.dart:18]: See: #772

Comment (matibat) [reply lib/utils/app_platform.dart]: > do we want to put the platform tag in there?

Yes we do!
